src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -565,9 +565,14 @@ private static Assembly SgenAssemblyResolver(object source, ResolveEventArgs e)
                     return null;
                 }
 
-                if(s_referencedic.ContainsKey(assemblyname))
+                if (s_referencedic.ContainsKey(assemblyname))
                 {
                     string reference = s_referencedic[assemblyname];
+
+                    //for System.ServiceModel.Primitives, we need to load its execution assembly rather than reference assembly
```

Comment:
Do we understand why we were trying to load the reference assembly before?

Reply:
Yes. We need to reflect over the developers classes to generate the code for the serializer. A serializer needed for a WCF service needs to examine attributes which are defined in S.SM.Primitives. The reflection API's for attributes cause the attribute to be instantiated which means we can't load the reference assembly. For the rest of the data types referenced, a reference assembly is sufficient as there's no instantiation required.

---
In reply to: [220660138](https://github.com/dotnet/corefx/pull/32403#discussion_r220660138) [](ancestors = 220660138)